src/app/Http/Controllers/Auth/LoginController.php

```diff
@@ -0,0 +1,130 @@
+<?php
+
+namespace NetworkRailBusinessSystems\UserLogin\Http\Controllers\Auth;
+
+use AnthonyEdmonds\GovukLaravel\Helpers\GovukPage;
+use AnthonyEdmonds\GovukLaravel\Helpers\GovukQuestion;
+use Illuminate\Contracts\View\View;
+use Illuminate\Database\Eloquent\Model;
+use Illuminate\Http\RedirectResponse;
+use Illuminate\Routing\Controller;
+use Illuminate\Support\Facades\Auth;
+use NetworkRailBusinessSystems\UserLogin\Http\Requests\Auth\LoginRequest;
+
+class LoginController extends Controller
+{
+    public function index(): View
+    {
+        $view = config('user-login.view');
+
+        $action = route('login');
+
+        $buttonLabel = 'Sign in';
+
+        $questions = [
+            [
+                'name' => 'username',
+                'label' => 'What is your username?',
+                'hint' => 'The username you use to access your Windows device, such as jdoe3.',
+            ],
+            [
+                'name' => 'password',
+                'label' => 'What is your password?',
+                'hint' => 'The password you use to access your Windows device.',
+            ],
+        ];
+
+        return $view === 'gov-uk-login'
+            ? GovukPage::questions(
+                'Sign in',
+                [
+                    GovukQuestion::input($questions[0]['label'], $questions[0]['name'])
+                        ->hint($questions[0]['hint'])
+                        ->width(20),
+
+                    GovukQuestion::input($questions[1]['label'], $questions[1]['name'])
+                        ->hint($questions[1]['hint'])
+                        ->width(20),
+                ],
+                $buttonLabel,
+                $action,
+                null,
+                'post',
+                "user-login::$view"
+            )
+            : view("user-login::$view", [
+                'questions' => $questions,
+                'action' => $action,
+                'buttonLabel' => $buttonLabel,
+            ]);
+    }
+
+    public function signIn(LoginRequest $request): RedirectResponse
+    {
+        $usernameKey = config('user-login.ldap-sync') === true ? 'samaccountname' : 'username';
+
+        $details = [
+            $usernameKey => strtolower($request->username),
+            'password' => $request->password,
+        ];
+
+        if ($this->syncExistingUser($details[$usernameKey]) === false) {
+            return $this->loginFailed($details[$usernameKey]);
+        }
+
+        return Auth::attempt($details, true) === true
+            ? $this->loginSucceeded()
+            : $this->loginFailed($details[$usernameKey]);
+    }
+
+    protected function syncExistingUser(string $username): bool
+    {
+        $model = config('user-login.model');
+
+        $mail = $model::mail($username);
```

Comment:
This should be more abstract; ideally we would like to move to syncing based on something like GUID from Ldap, so this could be something along the lines of `uniqueIdentifier`. The interface trait then works the same, but with that wider context of it being a function that returns a unique identifier from the Auth source which should be present in the database.

Reply:
Added more configuration options, now trait can return **uniqueIdentifier** either from local database or ldap depending on configuration